shared/src/transport/http.rs

```diff
@@ -125,9 +126,26 @@ impl BatchTransport for HttpTransport {
     }
 }
 
+/// Workaround for Erigon nodes, which encode each element of the Batch Response as a String rather than a deserializable JSON object
+#[derive(Debug, PartialEq, Clone, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum OutputOrString {
+    String(String),
+    Output(Output),
+}
+
+impl From<OutputOrString> for Output {
+    fn from(output_or_string: OutputOrString) -> Self {
+        match output_or_string {
+            OutputOrString::String(string) => jsonrpc_core::serde_from_str(&string).unwrap(),
```

Comment:
We shouldn't be unwrapping on data coming from outside. I think this would panic if the string is invalid json.

Reply:
Agreed. If you want to use conversion traits, you can also use the `TryFrom`. Although, because this is an internal type, I would just recommend a `try_into_output` member function.

Reply:
Changed to return an error and error the whole batch. The motivation behind erroring the whole batch is that it may be possible to not be able to line up message IDs with response IDs if it fails to parse string outputs for whatever reason.